File: 136.single-number.js
/*
 * @lc app=leetcode id=136 lang=javascript
 *
 * [136] Single Number
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
  let hashTable = {};
  for (let i = 0; i < nums.length; i++) {
    const current = nums[i];
    const existingPair = hashTable[current];
    if (existingPair) delete hashTable[current];
    else hashTable[nums[i]] = 1;
  }
  return Object.keys(hashTable)[0];
};
// @lc code=end

File: 136.single-number.java
/*
 * @lc app=leetcode id=136 lang=java
 *
 * [136] Single Number
 */

// @lc code=start

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int singleNumber(int[] nums) {

        Map hashTable = new HashMap();

        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];
            Integer existingPair = hashTable.get(current);

            if (existingPair != null) {
                hashTable.remove(current);
            } else {
                hashTable.put(nums[i], 1);
            }
        }

        System.out.println(hashTable);
        return hashTable;

    }
}
// @lc code=end
